### PR TITLE
Agentless CSPM Azure sanity tests

### PR DESCRIPTION
**Pull request by amirbenun**
### Summary of your changes
On this PR, the sanity tests install CSPM Azure integration on the agentless policy and verify it produces findings.

### Related Issues
- Resolves: https://github.com/elastic/cloudbeat/issues/1602

### Checklist
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

#### Introducing a new rule?

- [ ] Generate rule metadata using [this script](https://github.com/elastic/cloudbeat/tree/main/security-policies/dev#generate-rules-metadata)
- [ ] Add relevant unit tests
- [ ] Generate relevant rule templates using [this script](https://github.com/elastic/cloudbeat/tree/main/security-policies/dev#generate-rule-templates), and open a PR in [elastic/packages/cloud_security_posture](https://github.com/elastic/integrations/tree/main/packages/cloud_security_posture)

### CODE REVIEW

File: tests/integration/tests/test_sanity_checks.py [hunk 1]
```diff
@@ -239,14 +244,18 @@ def test_cspm_azure_findings(cspm_client, match_type):
...
-    )
+    azure_agents = wait_components_list(agents_actual_components, agents_expected_components, CIS_AZURE_COMPONENT)
+    for agent in azure_agents:
+        query_list = build_query_list(
+            benchmark_id="cis_azure",
+            version=AGENT_VERSION,
+            match_type=match_type,
+            agent=agent,
+        )
+        query, sort = cspm_client.build_es_must_match_query(must_query_list=query_list, time_range="now-24h")
```
Comment (oren-zohar) [inline tests/integration/tests/test_sanity_checks.py:255]: @amirbenun shouldn't we only check the agentless agent?

Comment (amirbenun) [reply tests/integration/tests/test_sanity_checks.py:255]: The former revision of this function asserts for any findings of this benchmark, after installing CSPM Azure on the agentless agent, there are 2 agents running this benchmark. To verify they both produce findings, I have added the for loop to assert separately.

**Comment by amirbenun** [comment]
Successful run: https://github.com/elastic/cloudbeat/actions/runs/7655445918

**Review by oren-zohar — APPROVED** [review]
Some actions show warning and need to be updated, but other than that it looks good

**Pull request merged**